src/QMCHamiltonians/tests/test_ion_derivs.cpp

```diff
@@ -772,4 +772,778 @@ TEST_CASE("Eloc_Derivatives:multislater_wj", "[hamiltonian]")
+
+//  REQUIRE(nlpp_obs == Approx(1.3849558361e+01));
+
+  ParticleSet::ParticleGradient_t fnlpp_complex(ions.getTotalNum());
+  ParticleSet::ParticlePos_t fnlpp(ions.getTotalNum());
+  for (int ionid = 0; ionid < ions.getTotalNum(); ionid++)
+  {
+    for (int idim = 0; idim < OHMMS_DIM; idim++)
+    {
+//      twf.wipe_matrix(dB[idim]);
+//      twf.wipe_matrix(dM[idim]);
+    }
+
+//    twf.get_igrad_M(elec, ions, ionid, dM);
+//    nlppop->evaluateOneBodyOpMatrixForceDeriv(elec, ions, twf, ionid, dB);
+
+    for (int idim = 0; idim < OHMMS_DIM; idim++)
+    {
+//      twf.get_gs_matrix(dB[idim], dB_gs[idim]);
+//      twf.get_gs_matrix(dM[idim], dM_gs[idim]);
+//      fnlpp_complex[ionid][idim] = twf.compute_gs_derivative(minv, X, dM_gs[idim], dB_gs[idim]);
+    }
+    convertToReal(fnlpp_complex[ionid], fnlpp[ionid]);
+  }
+
+#if defined(MIXED_PRECISION)
+//  REQUIRE(fnlpp[0][0] == Approx(24.2239540340527491).epsilon(2e-4));
+//  REQUIRE(fnlpp[0][1] == Approx(-41.9981344310649263).epsilon(2e-4));
+//  REQUIRE(fnlpp[0][2] == Approx(-98.9123955744908159).epsilon(2e-4));
+//  REQUIRE(fnlpp[1][0] == Approx(2.5105943834091704).epsilon(2e-4));
+//  REQUIRE(fnlpp[1][1] == Approx(1.1345766918857692).epsilon(2e-4));
+//  REQUIRE(fnlpp[1][2] == Approx(-5.2293234395150989).epsilon(2e-4));
+#else
+//  REQUIRE(fnlpp[0][0] == Approx(24.2239540340527491));
+//  REQUIRE(fnlpp[0][1] == Approx(-41.9981344310649263));
+//  REQUIRE(fnlpp[0][2] == Approx(-98.9123955744908159));
+//  REQUIRE(fnlpp[1][0] == Approx(2.5105943834091704));
+//  REQUIRE(fnlpp[1][1] == Approx(1.1345766918857692));
+//  REQUIRE(fnlpp[1][2] == Approx(-5.2293234395150989));
+#endif
+}
+
+/*TEST_CASE("Eloc_Derivatives:slater_wj", "[hamiltonian]")
+{
+  app_log() << "====Ion Derivative Test: Single Slater+Jastrow====\n";
+  using RealType = QMCTraits::RealType;
+
+  Communicate* c;
+  c = OHMMS::Controller;
+
+  ParticleSet ions;
+  ParticleSet elec;
+
+  create_CN_particlesets(elec, ions);
+
+  int Nions = ions.getTotalNum();
+  int Nelec = elec.getTotalNum();
+
+  HamiltonianFactory::PtclPoolType particle_set_map;
+  HamiltonianFactory::PsiPoolType psi_map;
+
+  particle_set_map["e"]    = &elec;
+  particle_set_map["ion0"] = &ions;
+
+  HamiltonianFactory hf("h0", elec, particle_set_map, psi_map, c);
+
+  WaveFunctionFactory wff("psi0", elec, particle_set_map, c);
+  psi_map["psi0"] = &wff;
+
+  const char* hamiltonian_xml = "<hamiltonian name=\"h0\" type=\"generic\" target=\"e\"> \
+         <pairpot type=\"coulomb\" name=\"ElecElec\" source=\"e\" target=\"e\"/> \
+         <pairpot type=\"coulomb\" name=\"IonIon\" source=\"ion0\" target=\"ion0\"/> \
+         <pairpot name=\"PseudoPot\" type=\"pseudo\" source=\"ion0\" wavefunction=\"psi0\" format=\"xml\" algorithm=\"non-batched\"> \
+           <pseudo elementType=\"C\" href=\"C.ccECP.xml\"/> \
+           <pseudo elementType=\"N\" href=\"N.ccECP.xml\"/> \
+         </pairpot> \
+         </hamiltonian>";
```

Comment:
Seems to be a good amount of repeated code here.  Refactor to some to test class at top of file or in separate header/implementation to avoid repetition?

Reply:
Good call.  I'll move the code into a function (like the CN particle set construction), and replace these code chunks throughout this source file.

Reply:
Done.